starknet_py/net/account/account.py

```diff
@@ -663,12 +651,7 @@ async def deploy_account(
             auto_estimate=auto_estimate,
         )
 
-        if chain in (
-            StarknetChainId.SEPOLIA_TESTNET,
-            StarknetChainId.SEPOLIA_INTEGRATION,
-            StarknetChainId.GOERLI,
-            StarknetChainId.MAINNET,
-        ):
+        if chain in StarknetChainId:
             balance = await account.get_balance()
             if balance < deploy_account_tx.max_fee:
```

Comment:
I am not sure how to implement a similar check for the V3 transaction now. I'm guessing eventually we'd need to pull the balance from the STRK contract and an exchange rate to WEI?

Let me know if I am missing something here. Generally I think it is not crucial, but definitely nice to have.

Reply:
Good question. I suppose it could depend on fee DA mode as well, but I'm not 100% sure. Let's open an issue to investigate this, we might need to ask SW as well.

Reply:
https://github.com/software-mansion/starknet.py/issues/1266

Reply:
This is not crucial since tx will be rejected with a descriptive enough error message anyway

> I'm guessing eventually we'd need to pull the balance from the STRK contract and an exchange rate to WEI?

It should probably be done the same way, except we'll have to pass STRK token address to `get_balance`. I don't really understand why'd you want to convert to WEI though?

Reply:
You're absolutely right, there is no need for conversion. I thought that `L1_GAS` in the resource bounds is specified in WEI, even though V3 transaction costs are paid in FRI, which is not the case. Apologies for the confusion.

We can indeed check it the same way as for V1, but with the STRK contract. On the other hand, as @DelevoXDG said, it is not crucial, since it will throw an error anyway. @THenry14 Wdyt?

Reply:
Agreed, let's update #1266 with the findings and do necessary changes under that issue at some later date